lib/index.js

```diff
@@ -100,21 +100,121 @@ Intercom.prototype.group = function(group, fn){
 };
 
 /**
- * Track the user's action
+ * Track the user's action via Bulk Event API
+ *
+ * https://developers.intercom.com/reference#bulk-event-operations
+ *
+ * 1) Lookup in redis if there is an existing job for a `userId`
+ * 2) If exists, add to the job
+ * 3) Otherwise create a new job, save jobId in redis with expiration
+ * 4) Error handle in case we try to add to an invalid job
+ *
+ * Note: Jobs stay open for 15 minutes but messages are processed right away
  *
  * @param {Track} track
  * @param {Function} fn
  * @api public
  */
 
-Intercom.prototype.track = function(track, fn){
+Intercom.prototype.track = function(event, fn){
+  var id = event.userId() || event.email();
+  // Adding 'events' so we can still lock separately from locking in `.identify()` calls
+  var key = [this.settings.appId, 'events', id].join(':');
```

Comment:
Yeah I had that in a draft review that I haven't published yet 😄
We should lock the same for these to ensure identify calls get made first, since the tracks could be run immediately.